public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/AddNotificationChannelModal.test.tsx

```diff
@@ -61,4 +105,45 @@ describe('AddNotificationChannelModal', () => {
 
     expect(wrapper.find(dataQa('name-text-input')).prop('value')).toEqual(notificationChannelStubs[0].summary);
   });
+
+  it('should have the submit button initially disabled', () => {
+    const wrapper = mount(<AddNotificationChannelModal setVisible={jest.fn()} isVisible />);
+    const button = findFormButton(wrapper);
+
+    expect(button.getDOMNode().hasAttribute('disabled')).toBe(true);
+  });
+
+  describe('Pager Duty option', () => {
+    it('should have the submit button disabled if both routing and service keys are empty', () => {
+      const wrapper = mount(<AddNotificationChannelModal setVisible={jest.fn()} isVisible />);
+      const button = findFormButton(wrapper);
+
+      fillNameAndType(wrapper, NotificationChannelType.pagerDuty);
+      expect(button.getDOMNode().hasAttribute('disabled')).toBe(true);
+    });
+
+    it('should enable submit button if only routing key is provided', () => {
+      const wrapper = mount(<AddNotificationChannelModal setVisible={jest.fn()} isVisible />);
+      const button = findFormButton(wrapper);
+
+      fillNameAndType(wrapper, NotificationChannelType.pagerDuty);
+
+      const routingKeyInput = wrapper.find(dataQa('routing-text-input')).at(0);
+
+      routingKeyInput.simulate('change', { target: { value: 'Sample routing key' } });
+      expect(button.getDOMNode().hasAttribute('disabled')).toBe(false);
```

Comment:
nit: `getDOMNode().hasAttribute()` seems a bit clunky and foreign to how we write tests. I'd say `.props(<propName>) should work OK for this purpose :)

Reply:
Done 👍